extensions/amp-sidebar/0.1/amp-sidebar.js

```diff
@@ -41,10 +42,148 @@ export class AmpSidebar extends AMP.BaseElement {
     /** @const @private {boolean} */
     this.isExperimentOn_ = isExperimentOn(this.getWin(), EXPERIMENT);
 
+    /** @private @const {!Window} */
+    this.win_ = this.getWin();
+
+    /** @private @const {!Document} */
+    this.document_ = this.win_.document;
+
+    /** @private @const {!Element} */
+    this.documentElement_ = this.document_.documentElement;
+
+    /** @private @const {string} */
+    this.side_ = this.element.getAttribute('side');
+
+    /** @private @const {!Viewport} */
+    this.viewport_ = this.getViewport();
+
+    /** @private @const {!Element} */
+    this.maskElement_ = false;
+
     if (!this.isExperimentOn_) {
       dev.warn(TAG, `Experiment ${EXPERIMENT} disabled`);
       return;
     }
+
+    if (this.side_ != 'left' && this.side_ != 'right') {
+      const pageDir =
+          this.document_.body.getAttribute('dir') ||
+          this.documentElement_.getAttribute('dir') ||
+          'ltr';
+      this.side_ = (pageDir == 'rtl') ? 'right' : 'left';
+      this.element.setAttribute('side', this.side_);
+    }
+
+    if (this.isOpen_()) {
+      this.open_();
+    } else {
+      this.element.setAttribute('aria-hidden', 'true');
+    }
+
+    this.documentElement_.addEventListener('keydown', event => {
+      // Close sidebar on ESC.
+      if (event.keyCode == 27) {
+        this.close_();
+      }
+    });
+
+    //TODO (skrish, #2712) Add history support on back button.
+    this.registerAction('toggle', this.toggle_.bind(this));
+    this.registerAction('open', this.open_.bind(this));
+    this.registerAction('close', this.close_.bind(this));
+  }
+
+ /**
+  * Returns true if the sidebar is opened.
+  * @returns {boolean}
+  * @private
+  */
+ isOpen_() {
+   return this.element.hasAttribute('open');
+ }
+
+  /** @override */
+  activate() {
+    this.open_();
+  }
+
+
+  /**
+   * Toggles the open/close state of the sidebar.
+   * @private
+   */
+  toggle_() {
+    if (this.isOpen_()) {
+      this.close_();
+    } else {
+      this.open_();
+    }
+  }
+
+  /**
+   * Reveals the sidebar.
+   * @private
+   */
+  open_() {
+    this.viewport_.disableTouchZoom();
+    this.mutateElement(() => {
+      this.viewport_.addToFixedLayer(this.element);
+      setStyles(this.element, {
```

Comment:
in this case the sidebar needs to be block; since it is a custom element. we would need to make it block